src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -107,6 +123,16 @@ public static class PackageDownloader
 
                     foreach (var source in sourceGroup)
                     {
+                        if (isPackageNamespaceEnabled)
+                        {
+                            if (configuredPackageSources != null &&
+                                !configuredPackageSources.Contains(source.PackageSource.Name.ToLowerInvariant()))
```

Comment:
With other recommendations, I think you can remove https://github.com/NuGet/NuGet.Client/pull/4039/files#diff-41725cea1cdd668d072407b362c5ce69b6e21ad72bb28b9499e897e11264b083R155.

Reply:
Replaced with `!configuredPackageSources.Contains(source.PackageSource.Name, StringComparer.CurrentCultureIgnoreCase)`.